### DISCUSSION TITLE
stuck and get a blank display on centos7

### DISCUSSION THREAD

**Discussion by caicaixuehua** [eteran/edb-debugger#813] [Q&A] [answered]
On centos7,the edb will stuck and the gui is blank,I don't know why.
the stucked line:
![85D7B0B1-5B28-4f7b-9130-9E8B04835904](https://user-images.githubusercontent.com/49799805/145387881-217b719a-27a4-408b-be45-2d0df5839b9a.png)
stucked gui:
![image](https://user-images.githubusercontent.com/49799805/145388107-67ea38aa-4aab-4582-8370-61245df20a47.png)

**Comment by caicaixuehua** [comment]
If I debug 32bit mode,it will not be stucked. If I debug 64bit mode,it will be stucked.What happened?

**Reply by caicaixuehua** [reply]
Debug 32bit mode program,edb also may be stucked.

**Reply by caicaixuehua** [reply]
When the gui is blank, botton of step into,step over is okay(the registers panel is okay and display rightly). Step out has the following error:
![image](https://user-images.githubusercontent.com/49799805/145511567-fc101b33-c1a4-4f01-bf4a-80c197e5c857.png)

**Comment by 10110111** [comment, marked as answer]
I think EDB doesn't support debugging across `execve` system call, which your screenshot is showing.

**Reply by caicaixuehua** [reply]
I don't want to debugging execve system call,but how can I skip it and then debug the following user program.

**Reply by 10110111** [reply]
You can't skip `execve`, you'll be inside another process after the call. But yeah, here's a patch that should help you get into that another process:
```diff
diff --git a/plugins/DebuggerCore/unix/linux/PlatformProcess.cpp b/plugins/DebuggerCore/unix/linux/PlatformProcess.cpp
index bda32ffb..771e6945 100644
--- a/plugins/DebuggerCore/unix/linux/PlatformProcess.cpp
+++ b/plugins/DebuggerCore/unix/linux/PlatformProcess.cpp
@@ -317,10 +317,12 @@ std::size_t PlatformProcess::readBytes(edb::address_t address, void *buf, std::s
 		if (readOnlyMemFile_) {
 			seek_addr(*readOnlyMemFile_, address);
 			read = readOnlyMemFile_->read(ptr, len);
-			if (read == 0 || read == quint64(-1)) {
+			if (read == quint64(-1)) {
 				return 0;
 			}
-		} else {
+		}
+		if (read == 0) {
+			// Fallback to PTRACE_PEEK
 			for (std::size_t index = 0; index < len; ++index) {
 
 				// read a byte, if we failed, we are done
```

**Reply by caicaixuehua** [reply]
Thank you,I will try it on Monday.

**Reply by caicaixuehua** [reply]
After I changed PlatformProcess.cpp,I don't get a blank gui,but I encountered this error when I use step out:
![image](https://user-images.githubusercontent.com/49799805/145805806-fc07f8e4-a2aa-4079-b61c-4986516fb97a.png)

I always use step out before this error,but I don't know how it enter ld-2.17.so.I don't want to enter ld-2.17.so.

**Reply by 10110111** [reply]
As I wrote above, you're now in another process, and `ld-2.17.so` is where its entry point is. There's no way around `execve`, it doesn't return unless it fails. The caller process image no longer exists at this point.

**Reply by caicaixuehua** [reply]
I think I know what you mean.Is there a way to debug the another process?

**Reply by 10110111** [reply]
You're already debugging it after you stepped into `syscall`...

**Reply by caicaixuehua** [reply]
Okay.But it is always in ld-2.17.so when I step over. Not enough step over?

**Reply by 10110111** [reply]
Of course, from `_start` to `__libc_start_main` to `main` you'd have to step quite a lot of steps if you do it manually. You should put a breakpoint at `main` or whatever you have, and run the process, as you normally do at the beginning of your debug sessions.

**Reply by caicaixuehua** [reply]
But how to find the main entry point?

**Reply by 10110111** [reply]
Apparently, EDB doesn't support this case well. In particular, it can't set breakpoints after `execve` or modify memory in any other way, so you're out of luck here.

The only suggestion I have is to start the new process manually in a separate debugging session with appropriate parameters and environment (this might not work if it relies on inherited descriptors, but well). Then _Function Finder_ can find `main`, you can set a breakpoint there etc.

**Reply by caicaixuehua** [reply]
Yes,it can't set breakpoints after execve.Okay, I know,thank you very much.

**Comment by 10110111** [comment]
@eteran
It seems `execve` invalidates `readOnlyMemFile_`. Dunno why though.

**Reply by eteran** [reply]
Very interesting! Will have to look into this for sure.
